src/GraphQL/Instrumentation/GraphQLTelemetryOptions.cs

```diff
@@ -0,0 +1,18 @@
+#if NET5_0_OR_GREATER
+
+namespace GraphQL.Instrumentation;
+
+/// <summary>
+/// Options for configuring the telemetry instrumentation.
+/// </summary>
+public class GraphQLTelemetryOptions
+{
+    // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/graphql.md
+
+    /// <summary>
+    /// Indicates whether the GraphQL document is included in the telemetry.
+    /// </summary>
+    public bool RecordDocument { get; set; } = true; // recommended
```

Comment:
Also I suggest to add 
1. `public Action<Activity, ExecutionOptions> EnrichWithExecutionOptions { get; set; }`
2. `public Action<Activity, Exception> EnrichWithException { get; set; }`
3. `public bool RecordException { get; set; }`

The idea to make behavior customizable. See `AspNetCoreInstrumentationOptions` sources and usages.

Reply:
`RecordDocument` is not a filter; it is a boolean which helps define how the `Activity` is enriched.

Once the other outstanding questions regarding are addressed within this PR, then we can circle back to adding more extensibility.

Reply:
Please note that the current design allows extensibility via a derived class -- `SetInitialTagsAsync`, `SetOperationTagsAsync` and `SetResultTagsAsync` are all `protected virtual`.

Reply:
I understand. My idea that it's better to follow MS patterns in tracing (`AspNetCoreInstrumentationOptions` as example). `SetInitialTagsAsync, SetOperationTagsAsync , SetResultTagsAsync ` methods smell IMO since all of them look like settings saying what tags to set.

Reply:
Also check https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md and `ActivityExtensions.RecordException`. I see instrumentation libraries declare
`  public bool RecordException { get; set; }` property in their options.

Reply:
> My idea that it's better to follow MS patterns in tracing

In general I agree, however, these are not MS' patterns.  These are OpenTelemetry's patterns.  I generally do not like delegates set in options classes; for one, this prevents them from being serializable.  But in certain cases they are fine.  There are certainly some MS samples of such usage, although I find this to be a rare occurrence.

Here, I would be okay with adding delegates so long as the delegates by default do nothing.  Otherwise it would be a real pain to extend functionality.  For example, I think this is a stupid design:

```cs
    .AddTelemetry(telemetryOptions =>
    {
        var baseEnrichWithExecutionOptions = telemetryOptions.EnrichWithExecutionOptions;
        telemetryOptions.EnrichWithExceptionOptions = (a, b) => {
            baseEnrichWithExecutionOptions(a, b);
            // do other stuff
        };
    }));
```

But so long as the delegates were additive, and so long as the class could be derived from in the case that async behavior was desired to be added, that's fine.

Reply:
>  generally do not like delegates set in options classes; for one, this prevents them from being serializable.

It's a known drawback. However, this is used almost everywhere - filtering, enriching, configuring - always, when the client has an unlimited or difficult predictable set of actions over the object of interest.

>  so long as the delegates by default do nothing

We may mix two worlds - provide empty delegates AND `bool RecordXXX` options so behavior can be configured both by serializable settings (simple case) and by delegates (advanced case).

Reply:
> We may mix two worlds...

Sure

Reply:
Regarding my suggestion about `public bool RecordException { get; set; }` property. I realized that if added that property requires to add some code and that code should either 1) call `Activity.RecordException` from `OpenTelemetry.Api` package, or 2) replicate its behavior via copy-pasted code in our codebase. I dislike both approaches. So the current impl (just empty delegate) may be the better way. I added suggestion to enrich xml comment with note about `Activity.RecordException`. Closing this conversation.